packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
       }
     } else if ("value" in desc && desc.writable) {
       desc.value = value;
```

Comment:
I am confused about this, though, but _hopefully_ this is not broken as it is here. Step `4.d.iv` does a `[[DefineOwnProperty]]` call, but this just directly updates the `.value` in the descriptor without calling `defineProperty`.

```js
var o = { foo: 'bar' }; Object.getOwnPropertyDescriptor(o, 'foo').value = 'baz'
```

is not sufficient to change the value of `o.foo`.